plugins/rocksdb/src/main/java/org/hyperledger/besu/plugin/services/storage/rocksdb/segmented/RocksDBColumnarKeyValueStorage.java

```diff
@@ -93,10 +96,33 @@ public RocksDBColumnarKeyValueStorage(
       final MetricsSystem metricsSystem,
       final RocksDBMetricsFactory rocksDBMetricsFactory)
       throws StorageException {
+    this(configuration, segments, List.of(), metricsSystem, rocksDBMetricsFactory);
+  }
+
+  public RocksDBColumnarKeyValueStorage(
+      final RocksDBConfiguration configuration,
+      final List<SegmentIdentifier> segments,
+      final List<SegmentIdentifier> ignorableSegments,
+      final MetricsSystem metricsSystem,
+      final RocksDBMetricsFactory rocksDBMetricsFactory)
+      throws StorageException {
 
     try (final ColumnFamilyOptions columnFamilyOptions = new ColumnFamilyOptions()) {
+      final List<SegmentIdentifier> trimmedSegments = new ArrayList<>(segments);
+      final List<byte[]> existingColumnFamilies =
+          RocksDB.listColumnFamilies(new Options(), configuration.getDatabaseDir().toString());
+      ignorableSegments.forEach(
+          ignore -> {
+            trimmedSegments.remove(ignore);
+            existingColumnFamilies.forEach(
+                existed -> {
+                  if (Arrays.equals(existed, ignore.getId())) {
+                    trimmedSegments.add(ignore);
```

Comment:
If we have a new database with no existing column families can the ignorable segment be used?

Reply:
yes, the logic is that only ignore given column family if 1, it is marked to be ignored (for example, chain pruning is disabled) and 2, it does not exist in the database. This means if besu starts from a new database and chain pruning is disabled, it will ignore and create/load a db with one less column family.

Reply:
There could be three cases in total:
1. There is chain pruning segment in the db already and regardless of whether chain pruning is enabled, the extra column family can't be ignored otherwise besu will fail to start.
2. There isn't chain pruning segment in the db and chain pruning is disabled by default, besu will ignore the extra column family and the db will be backward compatible. (This will be de default behaviour as chain pruning is hidden) 
3. There isn't chain pruning segment in the db but chain pruning is enabled, besu will permanently add the extra column family and the db will be backward incompatible.